src/compiler/sys.ts

```diff
@@ -152,11 +161,10 @@ namespace ts {
                         watcher.referenceCount += 1;
                         return;
                     }
-                    watcher = _fs.watch(
+                    watcher = fsWatchDirectory(
                         dirPath || ".",
-                        { persistent: true },
                         (eventName: string, relativeFileName: string) => fileEventHandler(eventName, relativeFileName, dirPath)
-                    );
+                    ) as DirectoryWatcher;

```

Comment:
What makes this cast valid?

Reply:
we are setting the refcount property on the watcher below. (this is same as previous wherein watcher returned any and hence no cast needed)